File: VB/CustomEditor/RepositoryItemMyCheckedComboBoxEdit.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsApplication1
	<UserRepositoryItem("Register")>
	Public Class RepositoryItemMyCheckedComboBoxEdit
		Inherits RepositoryItemCheckedComboBoxEdit

		Shared Sub New()
			Register()
		End Sub
		Public Sub New()
		End Sub

		Friend Const EditorName As String = "MyCheckedComboBoxEdit"

		Public Shared Sub Register()
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(MyCheckedComboBoxEdit), GetType(RepositoryItemMyCheckedComboBoxEdit), GetType(MyCheckedComboBoxEditViewInfo), New MyButtonEditPainter(), True))
        End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EditorName
			End Get
		End Property

		Public Overridable Function MyFormatEditValue(ByVal value As Object) As Object
			If value Is Nothing Then
				Return value
			End If
			Return value.ToString().Replace(Environment.NewLine, String.Format("{0} ", SeparatorChar))
		End Function

		Public Overridable Function MyParseEditValue(ByVal value As Object) As Object
			If value Is Nothing Then
				Return value
			End If
			Return value.ToString().Replace(String.Format("{0} ", SeparatorChar), Environment.NewLine)
		End Function



				Protected Overrides Sub RaiseFormatEditValue(ByVal e As ConvertEditValueEventArgs)
			MyBase.RaiseFormatEditValue(e)
			e.Value = MyFormatEditValue(e.Value)
			e.Handled = True
				End Sub



		Protected Overrides Sub RaiseParseEditValue(ByVal e As ConvertEditValueEventArgs)
			MyBase.RaiseParseEditValue(e)
			e.Value = MyParseEditValue(e.Value)
			e.Handled = True
		End Sub

	End Class
End Namespace

File: CS/CustomEditor/RepositoryItemMyCheckedComboBoxEdit.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;

namespace WindowsApplication1
{
    [UserRepositoryItem("Register")]
    public class RepositoryItemMyCheckedComboBoxEdit : RepositoryItemCheckedComboBoxEdit
    {
        static RepositoryItemMyCheckedComboBoxEdit()
        {
            Register();
        }
        public RepositoryItemMyCheckedComboBoxEdit() { }

        internal const string EditorName = "MyCheckedComboBoxEdit";

        public static void Register()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(MyCheckedComboBoxEdit),
                typeof(RepositoryItemMyCheckedComboBoxEdit), typeof(MyCheckedComboBoxEditViewInfo),
                new MyButtonEditPainter(), true, null));
        }
        public override string EditorTypeName
        {
            get { return EditorName; }
        }

        public virtual object MyFormatEditValue(object value)
        {
            if (value == null)
                return value;
            return value.ToString().Replace(Environment.NewLine, String.Format("{0} ", SeparatorChar));
        }

        public virtual object MyParseEditValue(object value)
        {
            if (value == null)
                return value;
            return value.ToString().Replace(String.Format("{0} ", SeparatorChar), Environment.NewLine);
        }


       
                protected override void RaiseFormatEditValue(ConvertEditValueEventArgs e) {
            base.RaiseFormatEditValue(e);
            e.Value = MyFormatEditValue(e.Value);
            e.Handled = true;
        }



        protected override void RaiseParseEditValue(ConvertEditValueEventArgs e) {
            base.RaiseParseEditValue(e);
            e.Value = MyParseEditValue(e.Value);
            e.Handled = true;
        }

        }
}